assets/src/xdai.js

```diff
@@ -1,6 +1,49 @@
 const NativeAsset = require('./NativeAsset');
+const IXDaiBridge = require('./abi/IXDaiBridge.json');
 
-module.exports = new NativeAsset({
+class XDaiNativeAsset extends NativeAsset {
+  constructor(props) {
+    super(props);
+    this.bridgeAddress = '0x7301cfa0e1756b71869e93d4e4dca5c7d0eb0aa6';
+  }
+
+  async scanBlocks(address, fromBlock, toBlock) {
+    await super.scanBlocks(address, fromBlock, toBlock);
+
+    const web3 = this.getWeb3();
+    const contract = new web3.eth.Contract(IXDaiBridge, this.bridgeAddress);
+    const events = await contract.getPastEvents('AffirmationCompleted', {
+      fromBlock,
+      toBlock
+    });
+    const filteredEvents = events.filter(event => event.returnValues.recipient.toLowerCase() === address.toLowerCase());
+
+    for (const event of filteredEvents) {
+      this.core.addHistoryEvent({
+        id: `${event.transactionHash}-${event.logIndex}`,
+        asset: this.id,
+        type: 'send',
```

Comment:
If you want, you could define your own "event type".

Of course, the UI modules (modern-ui) would need to be updated to display them.